cmake/compiler-flags.cmake

```diff
@@ -2,19 +2,9 @@ include(CheckCXXCompilerFlag)
 include(CpuMarch)
 include(CheckCXXCompilerFlagAndEnableIt)
 
-# yes, need to keep both the CMAKE_CXX_FLAGS and CMAKE_CXX_STANDARD.
-# with just the CMAKE_CXX_STANDARD, try_compile() breaks:
-#   https://gitlab.kitware.com/cmake/cmake/issues/16456
-# with just the CMAKE_CXX_FLAGS, 'bundled' pugixml breaks tests
-#   https://github.com/darktable-org/rawspeed/issues/112#issuecomment-321517003
-
-message(STATUS "Checking for -std=c++17 support")
 CHECK_CXX_COMPILER_FLAG("-std=c++17" COMPILER_SUPPORTS_CXX17)
```

Comment:
I think we can drop this whole block too?

Reply:
I guess you'd still want to check if this is supported, or you're thinking the prior min. compiler version check is already sufficient to cover this?

Reply:
It's basically a TOCTOU issue.
I'm not sure what we get from that check nowadays,
since CMake (should?) handles it for us.

Reply:
Sure. I'll remove it.